src/pluginextension.h

```diff
@@ -0,0 +1,191 @@
+/*
+ *      pluginextension.h - this file is part of Geany, a fast and lightweight IDE
+ *
+ *      Copyright 2023 The Geany contributors
+ *
+ *      This program is free software; you can redistribute it and/or modify
+ *      it under the terms of the GNU General Public License as published by
+ *      the Free Software Foundation; either version 2 of the License, or
+ *      (at your option) any later version.
+ *
+ *      This program is distributed in the hope that it will be useful,
+ *      but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *      GNU General Public License for more details.
+ *
+ *      You should have received a copy of the GNU General Public License along
+ *      with this program; if not, write to the Free Software Foundation, Inc.,
+ *      51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+/**
+ *  @file pluginextension.h
+ *  This file defines an interface allowing plugins to take over some of the
+ *  core functionality provided by Geany: autocompletion, calltips,
+ *  symbol goto, and types highlighting inside document.
+ **/
+
+#ifndef GEANY_PLUGIN_EXTENSION_H
+#define GEANY_PLUGIN_EXTENSION_H 1
+
+#include "document.h"
+
+
+G_BEGIN_DECLS
+
+/**
+ * Structure serving as an interface between plugins and Geany allowing
+ * plugins to inform Geany about what features they provide and allowing
+ * Geany to delegate its functionality to the plugins.
+ * 
+ * Depending on the functionality they provide, plugins should assign pointers
+ * to the functions implementing this interface to the appropriate members of
+ * the structure. Not all of the functions have to be implemented by the
+ * plugin - member pointers of unimplemented functions can be left to contain
+ * the @c NULL pointer.
+ * 
+ * Typically, functions from this interface come in pairs. Functions assigned to
+ * the members ending with @c _provided inform Geany whether the plugin
+ * implements the given feature for the passed document. Functions assigned
+ * to the members ending with @c _perform are called by Geany at appropriate
+ * moments to inform the plugin when to perform the given feature.
+ * 
+ * The extension is defined by the pointers in the PluginExtension structure and
+ * is registered in Geany using the @c plugin_extension_register() function.
+ * 
+ * @warning The API provided by this file is subject to change and should not be
+ * considered stable at this point. That said, it is highly probable that if
+ * a change of this API is required in the future, it will not be of a major
+ * nature and should not require major modifications of the affected plugins
+ * (e.g. added/modified parameter of a function and similar changes).
+ **/
+typedef struct
+{
+	/**
+	 * Pointer to function called by Geany to check whether the plugin
```

Comment:
I got that @elextr requested this kind of wording, but to me it sounds absolutely weird. I would expect simpler language like "callback called by Geany to …" or even just "called by Geany to…"

Reply:
The struct member is not the function, its a pointer to the function, it can point to any function, its not a fixed function.  The original description seemed to suggest that the functions had fixed names, whereas its the member pointers that have the fixed names, and it was very confusing.

Reply:
Perhaps confusing for you, but to me it's clear: This is a table of callbacks implemented by the plugin. It's obvious to me that the plugin can chose any name internally.

For reference, see `GeanyProxyFuncs` that we already have documented

```

/** Hooks that need to be implemented by every proxy
 *
 * @see geany_plugin_register_proxy() for a full description of the proxy mechanism.
 *
 * @since 1.26 (API 226)
 **/
struct GeanyProxyFuncs
{
	/** Called to determine whether the proxy is truly responsible for the requested plugin.
	 * A NULL pointer assumes the probe() function would always return @ref GEANY_PROXY_MATCH */
	gint		(*probe)     (GeanyPlugin *proxy, const gchar *filename, gpointer pdata);
	/** Called after probe(), to perform the actual job of loading the plugin */
	gpointer	(*load)      (GeanyPlugin *proxy, GeanyPlugin *subplugin, const gchar *filename, gpointer pdata);
	/** Called when the user initiates unloading of a plugin, e.g. on Geany exit */
	void		(*unload)    (GeanyPlugin *proxy, GeanyPlugin *subplugin, gpointer load_data, gpointer pdata);
};
``` 

Anyway I don't request a change here, just that it's worded strangely to my programmer eyes.

Reply:
See my comment to @techee, because somebody understands it they are the wrong person to judge how understandable it is to others, its a normal human cognitive bias that we all have and _really_ hard to avoid.

Perhaps my confusion was because of the use of the member name as if it actually was a function name in the original documentation, I havn't had a chance to re-read it after @techee updated it, maybe it will be clearer, but of course now _I_ also understand it, sigh.